stackslib/src/chainstate/burn/db/sortdb.rs

```diff
@@ -1808,6 +1816,30 @@ impl<'a> SortitionHandleTx<'a> {
 
         if cur_epoch.epoch_id >= StacksEpochId::Epoch30 {
             // Nakamoto blocks are always processed in order since the chain can't fork
+            // arbitrarily.
+            // However, two kinds of forks can happen:
+            // * one of a set of malleableized siblings can be confirmed (but they all have the
+            // same sighash and same height).
```

Comment:
The comment here is out of date.  The code will create as many malleablized blocks as it can by stripping off signatures from the signers signature vector, but each time it does so, it also increments the block version.  This way, the blocks that get stored have different block hashes as well as different signature hashes, but remain semantically equivalent.

This is done to stress-test the MARF'ed aspect of the chainstate -- all histories must be stored, and all API methods should take and act upon a given Stacks chain tip to load only blocks within the fork represented by the tip.